pkg/apis/vsphere/v1alpha1/vsphereclusterproviderspec_types.go

```diff
@@ -77,6 +77,27 @@ type VsphereClusterProviderSpec struct {
 	// ClusterConfiguration holds the cluster-wide information used during a
 	// kubeadm init call.
 	ClusterConfiguration kubeadmv1beta1.ClusterConfiguration `json:"clusterConfiguration,omitempty"`
+
+	// VmwareCloud describes the supported cloud providers to run VMC
+	VmwareCloud *VmwareCloudSpec `json:"vmwareCloud,omitempty"`
+}
+
+// VmwareCloudSpec describes the supported cloud providers
+type VmwareCloudSpec struct {
+	// AwsProvider specifies the information needed to run VMC on AWS
+	AwsProvider *AwsProviderSpec `json:"awsProviderSpec"`
```

Comment:
i could be the only me think that name of `AwsProviderSpec` is confusing, because there is `VsphereProviderSpec`, but those two are not on the same level,  `AwsProviderSpec` is a sub-configure spec, i would like to hear how others think, whether name like `AwsSpec` make sense, thanks.